mne/encoding/feature.py

```diff
@@ -0,0 +1,269 @@
+"""Classes for transforming, preprocessing, and fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy.sparse import spmatrix, csr_matrix
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Creates delayed versions of a time series.
+
+    Parameters
+    ----------
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means time points in the past,
+        positive means time points in the future. Default is no delays.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.0
+
+    Attributes
+    ----------
+    ``names_`` : ndarray, shape (n_features * n_delays, 2)
+        If fit, a collection of `(feature_name, delay)` pairs
+        corresponding to columns of the output of `transform`.
+    """
+    def __init__(self, delays=0., sfreq=1.):
+        delays, _, sfreq = _check_delayer_params(
+            delays, sfreq)
+        self.delays = delays
+        self.sfreq = sfreq
+
+    def fit(self, X, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+            Training data.
+        y : array
+            Target values.
+
+        Returns
+        -------
+        self
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+
+        Returns
+        -------
+        X_delayed : array, shape (n_times, n_features * n_delays)
+            The delayed features of X.
+        """
+        X_delayed = delay_time_series(X, delays=self.delays,
+                                      sfreq=self.sfreq)
```

Comment:
I think we should be able to have irregular time lags. In my experience, the hankel tricks works better with log-spaced lags (probably because of the 1/f autocorr)